### PR TITLE
[control-plane-manager] Etcd automatic defragmentation

### PR DESCRIPTION
**Pull request by name212**
## Description
Etcd stops working if it exceeds space quota.

## Why do we need it, and what problem does it solve?
Check etcd storage usage and if it less than 10% free space left then trigger defrag operation.

## Changelog entries

```changes
section: control-plane-manager
type: feature
summary: Etcd automatic defragmentation.
impact_level: low
```

### CODE REVIEW

File: modules/040-control-plane-manager/openapi/config-values.yaml [hunk 1]
```diff
@@ -205,6 +205,12 @@ properties:
     description: |
       `etcd` parameters.
     properties:
+      disableAutoDefragmentation:
```
Comment (konstantin-axenov) [inline modules/040-control-plane-manager/openapi/config-values.yaml:208]: We decided that usage of `enable` is preferable than `disable`. Change this option to `enableAutoDefragmentation`.

File: modules/040-control-plane-manager/hooks/etcd_defragmentation.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+		}
+
+		input.LogEntry.Warnf("Start defrag etcd instance '%s' %d/%d", instance.PodName, status.DbSize, instance.MaxDbSize)
+		_, err = etcdClient.Defragment(context.TODO(), instance.Endpoint)
+		if err != nil {
+			input.MetricsCollector.Inc("d8_etcd_defragmentation_failed_total", map[string]string{
+				"pod_name": instance.PodName,
+				"node":     instance.Node,
+			})
+			input.LogEntry.Errorf("Defrag etcd '%s' instance finished with err: %v", instance.PodName, err)
```
Comment (konstantin-axenov) [inline modules/040-control-plane-manager/hooks/etcd_defragmentation.go:81]: Don't we need to stop defrag operation on an etcd cluster if defrag of one of the members failed? For example, an etcd member was killed by oom.

Comment (name212) [reply modules/040-control-plane-manager/hooks/etcd_defragmentation.go:81]: yep

Comment (name212) [reply modules/040-control-plane-manager/hooks/etcd_defragmentation.go:81]: I forgot about it. Thank you

**Comment by name212** [comment]
We decided to give up automatic defragmentation

**Pull request closed without merging**